Tasks/DownloadPackageV1/packagebuilder.ts

```diff
@@ -0,0 +1,138 @@
+import * as tl from "vsts-task-lib/task";
+
+import { Package } from "./package";
+import { SingleFilePackage } from "./singlefilepackage";
+import { MultiFilePackage } from "./multifilepackage";
+import { WebApi } from "azure-devops-node-api";
+
+export class PackageUrlsBuilder {
+    private type: string;
+    private pattern: string[];
+    private packagingMetadataAreaId: string;
+    private packageProtocolDownloadAreadId: string;
+    private extension: string;
+    private feedConnection: WebApi;
+    private pkgsConnection: WebApi;
+    private packageProtocolAreaName: string;
+
+    private getRouteParams: (feedId: string, packageMetadata: any, fileMetadata: any) => any;
+
+    get Type() {
+        return this.type;
+    }
+
+    ofType(type: string): PackageUrlsBuilder {
+        this.type = type;
+        switch (this.type) {
+            case "NuGet":
+                this.packagingMetadataAreaId = "7A20D846-C929-4ACC-9EA2-0D5A7DF1B197";
+                this.packageProtocolDownloadAreadId = "6EA81B8C-7386-490B-A71F-6CF23C80B388";
+                this.packageProtocolAreaName = "NuGet";
+                this.extension = ".zip";
+                break;
+            case "Npm":
+                this.packagingMetadataAreaId = "7A20D846-C929-4ACC-9EA2-0D5A7DF1B197";
+                this.packageProtocolDownloadAreadId = "09A4EAFD-123A-495C-979C-0EDA7BDB9A14"; 
```

Comment:
So this worked for scoped & unscoped packages?

Reply:
For scoped packages it returns the package name, just the file name replaces the / with _